docs/asm-gke-terraform/README.md

```diff
@@ -0,0 +1,105 @@
+## Managed Anthos Service Mesh on GKE with Terraform 
+
+This is a general setup of a single cluster with Anthos Service Mesh using the native [GKE Hub Terraform resource](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/gke_hub_feature) for the examples in the `docs` folder.
+
+The following will set up a GKE cluster with Anthos Service Mesh (ASM) with a _Managed Control Plane_ installed. 
+
+The following services will be required for this: 
+* `mesh.googleapis.com`
+
+### Prerequisites
+* This tutorial is best suited for [Cloud Shell](https://shell.cloud.google.com), which comes ready with the Google Cloud SDK and Terraform.
+*  You can follow this setup with either a [new Google Cloud Project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project) or a pre-existing Google Cloud Project. It is recommended to create a new Google Cloud Project, for an easier cleanup.
+
+## Quickstart with Terraform
+In your Cloud Shell, follow the steps outlined:
+
+### 1. Enable the APIs listed above
+```
+gcloud services enable mesh.googleapis.com
```

Comment:
**Question:**
Is there a reason we don't enable this API via Terraform (similar to [this example](https://github.com/GoogleCloudPlatform/microservices-demo/blob/5de83be7c3bbcc48bf41185b29c16200bfee6b39/terraform/main.tf#L14))?

Reply:
this is for having a more consistent path with TF, gcloud, etc, the ideal paths. 
I could put it in, but i'd like to be explicit with this command

Reply:
Actually, it would be ideal to have this in TF :)

Same with KCC, we should do it in KCC :)

For an end user perspective, choose your path:
- TF only
- gcloud only
- KCC only

:)